src/main/java/yanagishima/controller/QueryHistoryController.java

```diff
@@ -29,7 +28,7 @@
   private final QueryService queryService;
 
   @DatasourceAuth
-  @RequestMapping(value = "/queryHistory", method = { RequestMethod.GET, RequestMethod.POST })
+  @GetMapping("/queryHistory")
```

Comment:
We shouldn't use `@RequestMapping` if it's unused.
https://find-sec-bugs.github.io/bugs.htm#SPRING_CSRF_UNRESTRICTED_REQUEST_MAPPING